src/software/utils/main_imageProcessing.cpp

```diff
@@ -0,0 +1,251 @@
+
+#include <aliceVision/image/all.hpp>
+#include <aliceVision/system/cmdline.hpp>
+#include <aliceVision/system/Logger.hpp>
+#include <aliceVision/numeric/numeric.hpp>
+#include <aliceVision/sfmData/SfMData.hpp>
+#include <aliceVision/sfmDataIO/sfmDataIO.hpp>
+#include <aliceVision/config.hpp>
+
+#include <opencv2/opencv.hpp>
+#include <opencv2/core/core.hpp>
+#include <opencv2/core/eigen.hpp>
+#include <opencv2/imgproc.hpp>
+
+#include <OpenImageIO/imageio.h>
+#include <OpenImageIO/imagebuf.h>
+#include <OpenImageIO/imagebufalgo.h>
+
+#include <boost/program_options.hpp>
+#include <boost/filesystem.hpp>
+
+
+
+// These constants define the current software version.
+// They must be updated when the command line is changed.
+#define ALICEVISION_SOFTWARE_VERSION_MAJOR 2
+#define ALICEVISION_SOFTWARE_VERSION_MINOR 0
+
+using namespace aliceVision;
+namespace po = boost::program_options;
+namespace fs = boost::filesystem;
+
+
+int main(int argc, char * argv[]) {
+  std::string verboseLevel = system::EVerboseLevel_enumToString(system::Logger::getDefaultVerboseLevel());
+  std::string sfmInputDataFilename = "";
+  std::string sfmOutputDataFilepath = "";
+  bool reconstructedViewsOnly = false;
+  bool exposureCompensation = false;
+  float downscale = 1.0f;
+  float contrast = 1.0f;
+  int medianFilter = 0;
+  std::string extension;
+
+  int sharpenWidth = 1;
+  float sharpenContrast = 1.f;
+  float sharpenThreshold = 0.f;
+
+  // Command line parameters
+  po::options_description allParams(
+    "Parse external information about cameras used in a panorama.\n"
+    "AliceVision PanoramaExternalInfo");
+
+  po::options_description requiredParams("Required parameters");
+  requiredParams.add_options()
+    ("input,i", po::value<std::string>(&sfmInputDataFilename)->required(), "SfMData file input.")
+    ("outSfMData,o", po::value<std::string>(&sfmOutputDataFilepath)->required(), "SfMData file output.")
+    ;
+
+  po::options_description optionalParams("Optional parameters");
+  optionalParams.add_options()
+      ("reconstructedViewsOnly", po::value<bool>(&reconstructedViewsOnly)->default_value(reconstructedViewsOnly),
+       "Process only recontructed views or all views.")
+      ("downscale", po::value<float>(&downscale)->default_value(downscale),
+       "Downscale Factor (1.0: no change).")
+
+      ("exposureCompensation", po::value<bool>(&exposureCompensation)->default_value(exposureCompensation),
+       "Exposure Compensation.")
+
+      ("contrast", po::value<float>(&contrast)->default_value(contrast),
+       "Contrast Factor (1.0: no change).")
```

Comment:
for numeric params it would be better to add the range of values that can be taken with a brief explanation, i.e. [1 10], no change - max enhancement) to give at least a hint to the user on how to use it